parsers/UY.py

```diff
@@ -46,59 +64,98 @@ def get_salto_grande(session: Optional[Session]) -> float:
+        # get generation data
+        datum = {
+            key: parse_num(hour.find(spanish).contents[0])
+            for key, spanish in MAP_GENERATION.items()
+        }
+        # solar can sometimes return -0.1 at night, round up to 0
+        datum["solar"] = max(datum["solar"], 0)
+
+        # ingest date field
+        datefield = hour.find("hora").contents[0]
+        datestr = re.findall("\d\d/\d\d/\d\d\d\d \d+:\d\d", str(datefield))[0]
+        date = arrow.get(datestr, "DD/MM/YYYY h:mm").replace(
+            tzinfo=dateutil.tz.gettz(tz)
+        )
+        datum["time"] = date
+
+        # add to list
+        hour_recs.append(datum)
+    return hour_recs
+
+
+def fetch_consumption(
+    zone_key: str = "UY",
+    session: Optional[Session] = None,
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+) -> dict:
+    if target_datetime:
+        raise NotImplementedError("This parser is not yet able to parse past dates")
 
-    datefield = soup.find(
-        "span", attrs={"id": "ctl00_ContentPlaceHolder1_lblUltFecScada"}
-    )
-    datestr = re.findall("\d\d/\d\d/\d\d\d\d \d+:\d\d", str(datefield.contents[0]))[0]
-    date = arrow.get(datestr, "DD/MM/YYYY h:mm").replace(tzinfo=dateutil.tz.gettz(tz))
-
-    table = soup.find(
-        "table", attrs={"id": "ctl00_ContentPlaceHolder1_gridPotenciasNivel1"}
-    )
-
-    obj = {"datetime": date.datetime}
-
-    for tr in table.find_all("tr"):
-        tds = tr.find_all("td")
-        if not len(tds):
-            continue
-
-        key = tds[0].find_all("b")
-        # Go back one level up if the b tag is not there
-        if not len(key):
-            key = tds[0].find_all("font")
-        k = key[0].contents[0]
-
-        value = tds[1].find_all("b")
-        # Go back one level up if the b tag is not there
-        if not len(value):
-            value = tds[1].find_all("font")
-        v_str = value[0].contents[0]
-        if v_str.find(",") > -1 and v_str.find(".") > -1:
-            # there can be values like "1.012,5"
-            v_str = v_str.replace(".", "")
-            v_str = v_str.replace(",", ".")
-        else:
-            # just replace decimal separator, like "125,2"
-            v_str = v_str.replace(",", ".")
-        v = float(v_str)
-
-        # solar reports -0.1 at night, make it at least 0
-        v = max(v, 0)
-
-        obj[k] = v
+    entry = parse_page(session)[-1]  # get most recent timestamp
 
     # https://github.com/tmrowco/electricitymap/issues/1325#issuecomment-380453296
-    salto_grande = get_salto_grande(session)
-    obj["Hidráulica"] = obj.get("Hidráulica", 0.0) + salto_grande
+    entry = correct_for_salto_grande(entry, session)
 
-    return obj
+    data = {
+        "zoneKey": zone_key,
+        "datetime": entry["time"].datetime,
+        "consumption": entry["demand"],
+        "source": "ute.com.uy",
+    }
+
+    return data
 
 
 def fetch_production(
```

Comment:
Same as above, if you add `Session()` as the default then we can remove the optional types and just use required types.